planner/core/rule_column_pruning.go

```diff
@@ -166,14 +166,19 @@ func (p *LogicalUnionAll) PruneColumns(parentUsedCols []*expression.Column) erro
 	if !hasBeenUsed {
 		parentUsedCols = make([]*expression.Column, len(p.schema.Columns))
 		copy(parentUsedCols, p.schema.Columns)
+	} else {
+		for i := len(used) - 1; i >= 0; i-- {
```

Comment:
Add a comment here to inform that we do not directly use schema of children[0] because the column name may be changed?